### PR TITLE
Handle exception for the updation of a replication rule

### PR DESCRIPTION
**Pull request by skumarp7**
### Description
This PR handles the exception handling for updation of a replication rule. 

**Earlier behavior:** 

**First hit of the replication rule:** 
```
curl -Method POST -Uri "http://localhost:9201/_plugins/_replication/_autofollow?pretty"  -H @{'Content-Type' = 'application/json'} -body '{"leader_alias":"leader", "name": "test3", "
pattern":"test*"}'


StatusCode        : 200
StatusDescription : OK
Content           : {
                      "acknowledged" : true
                    }

```
**Second hit of the same Replication rule:** 
```
curl -Method POST -Uri "http://localhost:9201/_plugins/_replication/_autofollow?pretty"  -H @{'Content-Type' = 'application/json'} -body '{"leader_alias":"leader", "name": "test3", "
pattern":"test*"}'


StatusCode        : 200
StatusDescription : OK
Content           : {
                      "acknowledged" : true
                    }

```

**New behavior: 
First hit of the replication rule:** 
```
curl -Method POST -Uri "http://localhost:9201/_plugins/_replication/_autofollow?pretty"  -H @{'Content-Type' = 'application/json'} -body '{"leader_alias":"leader", "name": "test3", "
pattern":"test*"}'


StatusCode        : 200
StatusDescription : OK
Content           : {
                      "acknowledged" : true
                    }

```
**Second hit of the Replication rule:** 
```
curl -Method POST -Uri "http://localhost:9201/_plugins/_replication/_autofollow?pretty"  -H @{'Content-Type' = 'application/json'} -body '{"leader_alias":"leader", "name": "test3", "
pattern":"log*"}' 

curl : { "error" : { "root_cause" : [ { "type" : "resource_already_exists_exception", "reason" : "task with id {autofollow:leader:test3} already exist" } ], "type" : "exception", "reason" : "Exisiting Autofollow replication     
    rule cannot be updated", "caused_by" : { "type" : "resource_already_exists_exception", "reason" : "task with id {autofollow:leader:test3} already exist" } }, "status" : 500 }
```

 
### Issues Resolved
https://github.com/opensearch-project/cross-cluster-replication/issues/1394
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by skumarp7** [comment]
Hi @ankitkala & @monusingh-1 , 

Please provide feedback :)

**Comment by skumarp7** [comment]
Hi , 

I could see the integTest is failing for the below test case - 

```
org.opensearch.replication.integ.rest.StartReplicationIT > test operations are fetched from lucene when leader is in mixed mode FAILED
    org.opensearch.client.ResponseException: method [PUT], host [http://[::1]:34485], URI [_cluster/settings], status line [HTTP/1.1 400 Bad Request]
    {"error":{"root_cause":[{"type":"settings_exception","reason":"persistent setting [migration.direction], not recognized"}],"type":"settings_exception","reason":"persistent setting [migration.direction], not recognized"},"status":400}
        at __randomizedtesting.SeedInfo.seed([1A13834D768F0352:4342CA6FD3CA47EA]:0)
        at app//org.opensearch.client.RestClient.convertResponse(RestClient.java:385)
        at app//org.opensearch.client.RestClient.performRequest(RestClient.java:355)
        at app//org.opensearch.client.RestClient.performRequest(RestClient.java:330)
        at app//org.opensearch.replication.integ.rest.StartReplicationIT.test operations are fetched from lucene when leader is in mixed mode(StartReplicationIT.kt:1287)
REPRODUCE WITH: ./gradlew ':integTest' --tests "org.opensearch.replication.integ.rest.StartReplicationIT.test operations are fetched from lucene when leader is in mixed mode" -Dtests.seed=1A13834D768F0352 -Dtests.security.manager=true -Dtests.locale=smn-FI -Dtests.timezone=US/Samoa -Druntime.java=11

```

I was able to find a PR which involves the exclusion of this test case in integTestRemote Task but not in integTest. 

https://github.com/opensearch-project/cross-cluster-replication/pull/1379

File: build.gradle [hunk 1]
```diff
@@ -527,7 +527,9 @@ integTest {
             }
         }
     }
-
+    filter {
```
Comment (ankitkala) [inline build.gradle:530]: Why is this change required here?

Comment (skumarp7) [reply build.gradle:530]: The integTests were failing for few of them and i could find the below PR which involves the exclusion of this test case in integTestRemote Task but not in integTest.

https://github.com/opensearch-project/cross-cluster-replication/pull/1379

I have for now excluded this particular test case from the integTest to proceed for now. Let me know if any other plan to fix this test case as I can remove them from my current PR?

Comment (ankitkala) [reply build.gradle:530]: the test was disabled for remoteIntegTest due to a different issue. 
@astute-decipher did we cleanup the setting?

Comment (skumarp7) [reply build.gradle:530]: @ankitkala Shall I remove that filter then?

Comment (ankitkala) [reply build.gradle:530]: yes

Comment (skumarp7) [reply build.gradle]: done

**Review by ankitkala — APPROVED** [review]

**Review by monusingh-1 — APPROVED** [review]
